Makefile

```diff
@@ -29,16 +29,36 @@ build:
 
 check-dependencies:
 	@echo "$(YELLOW)Checking dependencies...$(RESET)"
+	@$(MAKE) -s check-system
 	@$(MAKE) -s check-python
 	@$(MAKE) -s check-npm
 	@$(MAKE) -s check-docker
 	@$(MAKE) -s check-poetry
 	@echo "$(GREEN)Dependencies checked successfully.$(RESET)"
 
+check-system:
+	@echo "$(YELLOW)Checking system...$(RESET)"
+	@if [ "$(shell uname)" = "Darwin" ]; then \
+		echo "$(BLUE)macOS detected.$(RESET)"; \
+	elif [ "$(shell uname)" = "Linux" ]; then \
+		echo "$(BLUE)Linux detected.$(RESET)"; \
+	elif [ "$$(uname -r | grep -i microsoft)" ]; then \
+		echo "$(BLUE)Windows Subsystem for Linux detected.$(RESET)"; \
+	else \
+		echo "$(RED)Unsupported system detected. Please use macOS, Linux, or Windows Subsystem for Linux (WSL).$(RESET)"; \
+		exit 1; \
+	fi
+		
 check-python:
 	@echo "$(YELLOW)Checking Python installation...$(RESET)"
 	@if command -v python3 > /dev/null; then \
-		echo "$(BLUE)$(shell python3 --version) is already installed.$(RESET)"; \
+		PYTHON_VERSION=$(shell python3 --version 2>&1 | awk '{print $$2}' | cut -d'.' -f1-2); \
```

Comment:
we got before some issues with the awk #824 and we removed this approach maybe if you can find some alternatives would be better

Reply:
how about using sed.

Reply:
Good catch! `sed` should work better, but be careful as some flags are different on OSX vs Linux

Reply:
Do we run `make build` in the OSS and Linux tests?

Reply:
Do you mean checking the system at the start of `make build` process?

Reply:
| Do we run make build in the OSS and Linux tests?
@rbren what do you mean exactly?

Reply:
I answered my own question--we do run `make build`

https://github.com/OpenDevin/OpenDevin/blob/main/.github/workflows/build_run-tests.yml

Reply:
Ahh got it 😅